File: src/test/java/com/everyonewaiter/domain/health/ApkVersionTest.java
package com.everyonewaiter.domain.health;

import static com.everyonewaiter.domain.health.ApkVersionFixture.createApkVersionCreateRequest;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class ApkVersionTest {

  @Test
  void create() {
    ApkVersionCreateRequest createRequest = createApkVersionCreateRequest();

    ApkVersion apkVersion = ApkVersion.create(createRequest);

    assertThat(apkVersion.getMajorVersion()).isEqualTo(createRequest.majorVersion());
    assertThat(apkVersion.getMinorVersion()).isEqualTo(createRequest.minorVersion());
    assertThat(apkVersion.getPatchVersion()).isEqualTo(createRequest.patchVersion());
    assertThat(apkVersion.getDownloadUri()).isEqualTo(createRequest.downloadUri());
  }

}

File: src/test/java/com/everyonewaiter/domain/health/ApkVersionTest.kt
package com.everyonewaiter.domain.health

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class ApkVersionTest {

  @Test
  fun `APK 버전 생성`() {
    val createRequest = createApkVersionCreateRequest()

    val apkVersion = ApkVersion.create(createRequest)

    Assertions.assertThat(apkVersion.majorVersion).isEqualTo(createRequest.majorVersion)
    Assertions.assertThat(apkVersion.minorVersion).isEqualTo(createRequest.minorVersion)
    Assertions.assertThat(apkVersion.patchVersion).isEqualTo(createRequest.patchVersion)
    Assertions.assertThat(apkVersion.downloadUri).isEqualTo(createRequest.downloadUri)
  }
}
